File: ASP.NET MVC. RedirectToAction propagando el QueryString/[C#]-ASP.NET MVC. RedirectToAction propagando el QueryString/C#/RedirectQueryString/Code/PropagarQueryStringAttribute.cs
using System;
using System.Collections.Generic;
using System.Collections.Specialized;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace RedirectQueryString.Code
{
    public class PropagarQueryStringAttribute: ActionFilterAttribute
    {
        public override void OnActionExecuted(ActionExecutedContext filterContext)
        {
            var redirectResult = filterContext.Result as RedirectToRouteResult;
            if (redirectResult != null)
            {
                NameValueCollection query = filterContext.HttpContext.Request.QueryString;
                foreach (string key in query.Keys)
                {
                    if (!redirectResult.RouteValues.ContainsKey(key))
                        redirectResult.RouteValues.Add(key, query[key]);
                }
            }
            base.OnActionExecuted(filterContext);
        }
    }
}

File: ASP.NET MVC. RedirectToAction propagando el QueryString/[VB]-ASP.NET MVC. RedirectToAction propagando el QueryString/VB/RedirectQueryStringVB/Code/PropagarQueryStringAttribute.vb
Imports System.Web.Mvc

Public Class PropagarQueryStringAttribute
    Inherits ActionFilterAttribute

    Public Overrides Sub OnActionExecuted(filterContext As ActionExecutedContext)
        Dim redirectResult As RedirectToRouteResult = filterContext.Result
        If (redirectResult IsNot Nothing) Then
            Dim query As NameValueCollection = filterContext.HttpContext.Request.QueryString
            For Each key As String In query.Keys
                If (Not redirectResult.RouteValues.ContainsKey(key)) Then
                    redirectResult.RouteValues.Add(key, query(key))
                End If
            Next
        End If
        MyBase.OnActionExecuted(filterContext)
    End Sub

End Class
